lib/dolly/document.rb

```diff
@@ -125,11 +125,15 @@ def self.write_methods name
     end
 
     def write_property name, value
+      instance_variable_set(:"@#{name}", value)
       @doc[name.to_s] = self.properties[name].value = value
     end
 
     def read_property name
-      doc[name.to_s] || self.properties[name].value
+      unless instance_variable_get(:"@#{name}")
+        instance_variable_set(:"@#{name}", (doc[name.to_s] || self.properties[name].value))
```

Comment:
`|| self.properties[name].value)` this is the one that will retrieve the default value?

also, please use write attribute, instead of repeating the instance var set